mmv1/products/spanner/api.yaml

```diff
@@ -102,8 +102,19 @@ objects:
         required: true
       - !ruby/object:Api::Type::Integer
         name: 'nodeCount'
-        description: 'The number of nodes allocated to this instance.'
-        default_value: 1
```

Comment:
Drive-by note: While removing a default is a safe change, that's **only** if the field is moved to `Optional` + `Computed`. We may need to use a `CustomizeDiff` to simulate a default of `1` when `processing_units` is unset, as breaking a config like the following working would also be backwards-incompatible. (Changing API requirements make backwards compatibility difficult!)

```
resource "google_spanner_instance" "example" {
  config       = "regional-us-central1"
  display_name = "Test Spanner Instance"
  labels = {
    "foo" = "bar"
  }
}
```

Reply:
How about if:
- I write an encoder that sets num_nodes to 1 if processing_units is 0, mark num_nodes as O+C and remove the default.
- Open an issue to make both of the fields ExactlyOneOf in v4 and remove the encoder?
- File an issue to fix that bug in the ruby generator.

Reply:
Yeah, that should also work. We should make sure to test a config w/o an explicit `node_count` if we don't already.